src/state/notifications/sagas.ts

```diff
@@ -109,7 +111,16 @@ export function* authenticateEmailSaga(action: AuthenticateEmailAction) {
     }
   } catch (error) {
     const { msg } = error.response.data;
-    yield put(authenticateEmailFailure(msg));
+    let readableErrorMessage = '';
+    switch (msg) {
+      case NotificationApiErrorMessages.WALLET_ALREADY_SUBSCRIBED:
+        readableErrorMessage = 'Wallet is already subscribed';
```

Comment:
not responsibility of saga to care about those errors, the mapping should be done in the selector, check enhancedError in https://github.com/bitcoinvault/GoldWallet/pull/597/files